src/main/java/com/savvato/tribeapp/controllers/ConnectAPIController.java

```diff
@@ -74,11 +74,17 @@ public ResponseEntity getQrCodeString(
   @Connect
   @PostMapping
   public boolean connect(@RequestBody @Valid ConnectRequest connectRequest) {
-    if (connectService.validateQRCode(connectRequest.qrcodePhrase, connectRequest.toBeConnectedWithUserId)) {
-      return connectService.saveConnectionDetails(connectRequest.requestingUserId, connectRequest.toBeConnectedWithUserId);
-    } else {
+
+    if (!connectService.validateQRCode(connectRequest.qrcodePhrase, connectRequest.toBeConnectedWithUserId)) {
```

Comment:
Let's move these to the beginning of saveConnectionDetails(). There was some validation code there which I think you removed in favor of this here, but let's replace that removed code with these validate*() calls.

Reply:
moving the QR code validation check to the service required changing the method signature parameter to Connection request. I deprecated saveConnectionDetails and created/integrated saveConnectionRequestDetails. Once I fix all the tests, I think I can safely delete the old method, but I'll check everything to make sure.

Reply:
Let's just use the same signature as was there before. You can pass in the userIds from the request. No need for the new method / deprecating the old one.

Reply:
Okay, to clarify:
option 1: 
- same name, same parameters: saveConnectionDetails(connectionRequest.requestingUserId, connectionRequest.toBeConnectedWithId)
- keep validate QR code in controller
- move the validate connections from controller to service
option 2: 
- same name, different parameter: saveConnectionDetails(connectionRequest)
- move all validation from controller to service

Reply:
Option 3:
- same name, same parameters: saveConnectionDetails(connectionRequest.requestingUserId, connectionRequest.toBeConnectedWithId)
- move all validation from controller to service

Reply:
Option 3:
- same name, same parameters: saveConnectionDetails(connectionRequest.requestingUserId, connectionRequest.toBeConnectedWithId)
- move all validation from controller to service

Reply:
The validateQRCode method in the ConnectService requires a QRcode phrase, which is one of the fields in the connectionRequest dto. If we move all validation to the connectService, we need to pass in the QRCode phrase. 

From what I can tell, option 2 above would accomplish this.

```
public Boolean validateQRCode(String qrcodePhrase, Long toBeConnectedWithUserId) {
        return qrcodePhrase.equals(getQRCodeString(toBeConnectedWithUserId).orElse("")) && StringUtils.isNotBlank(qrcodePhrase);
    }
```

Reply:
per meeting conversation, using option 2